`tkn describe pipelinerun` crashes with a Condition

I have been seeing `tkn` crash after adding a `Condition` to a `Pipeline`.  It's not 100% of the time:
- it seems worse right after creating the `PipelineRun`, and then may eventually settle out.
- it seems like a failed `Task` can make it never recover.

The backtraces match in terms of files & lines indicated.  The one below is representative.

# Version and Operating System

**tkn Version:** 0.8.0

**Operating System:** Debian buster

# Expected Behavior
not crashing :)

# Actual Behavior

```
$ tkn pipelinerun describe hello-hello --context tekton
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1130806]

goroutine 1 [running]:
github.com/tektoncd/cli/pkg/helper/pipelinerun/description.taskrunList.Less(0xc00011a540, 0x3, 0x4, 0x2, 0x1, 0xc00002da00)
        /workspace/src/github.com/tektoncd/cli/pkg/helper/pipelinerun/description/description.go:85 +0x66
sort.insertionSort(0x175a960, 0xc0004133c0, 0x0, 0x3)
        /usr/local/go/src/sort/sort.go:27 +0xc4
sort.quickSort(0x175a960, 0xc0004133c0, 0x0, 0x3, 0x4)
        /usr/local/go/src/sort/sort.go:209 +0x201
sort.Sort(0x175a960, 0xc0004133c0)
        /usr/local/go/src/sort/sort.go:218 +0x79
github.com/tektoncd/cli/pkg/helper/pipelinerun/description.PrintPipelineRunDescription(0xc00002dd40, 0x7fff033ea800, 0xb, 0x1772200, 0xc0002f6040, 0x0, 0x0)
        /workspace/src/github.com/tektoncd/cli/pkg/helper/pipelinerun/description/description.go:126 +0x883
github.com/tektoncd/cli/pkg/cmd/pipelinerun.describeCommand.func1(0xc000314dc0, 0xc00032ae40, 0x1, 0x3, 0x0, 0x0)
        /workspace/src/github.com/tektoncd/cli/pkg/cmd/pipelinerun/describe.go:73 +0x20b
github.com/spf13/cobra.(*Command).execute(0xc000314dc0, 0xc00032adb0, 0x3, 0x3, 0xc000314dc0, 0xc00032adb0)
        /workspace/pkg/mod/github.com/chmouel/cobra@v0.0.0-20200107083527-379e7a80af0c/command.go:834 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002ee2c0, 0xc0002f6040, 0xc0002ee2c0, 0xc0000a0058)
        /workspace/pkg/mod/github.com/chmouel/cobra@v0.0.0-20200107083527-379e7a80af0c/command.go:922 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
        /workspace/pkg/mod/github.com/chmouel/cobra@v0.0.0-20200107083527-379e7a80af0c/command.go:872
main.main()
        /workspace/src/github.com/tektoncd/cli/cmd/tkn/main.go:29 +0x4b
```

# Steps to Reproduce the Problem

1. `kubectl apply` the config below (sorry for the "minimal" config - this produces the most robust failures.)
1. `tkn pipelinerun describe hello-hello`

```yaml
---
apiVersion: tekton.dev/v1beta1
kind: Task
metadata:
  name: echo-hello-world
spec:
  params:
    - name: greeting
      type: string
  steps:
    - name: echo
      image: debian:buster-slim
      command:
        - echo
      args:
        - "Hello World: $(params.greeting)"
---
apiVersion: tekton.dev/v1beta1
kind: Task
metadata:
  name: pause
spec:
  params:
    - name: duration
      type: string
  steps:
    - name: echo
      image: debian:buster-slim
      command:
        - echo
      args:
        - "Sleeping for $(params.duration)"
    - name: sleep
      image: debian:buster-slim
      command:
        - sleep "$(params.duration)"   # mistake included on purpose

apiVersion: tekton.dev/v1beta1
kind: Pipeline
metadata:
  name: hello-hello
spec:
  params:
    - name: first-hello
      type: string
      default: wooo
    - name: second-hello
      type: string
      default: yeah
    - name: third-hello
      type: string
      default: heyy
    - name: do-extra-hello
      type: string
      default: "false"
  tasks:
    - name: first-hello
      taskRef:
        name: echo-hello-world
      params:
        - name: greeting
          value: "$(params.first-hello)"

    - name: pause-after-first
      taskRef:
        name: pause
      params:
        - name: duration
          value: "60"

    - name: second-hello
      taskRef:
        name: echo-hello-world
      params:
        - name: greeting
          value: "$(params.second-hello)"
      runAfter:
        - first-hello

    - name: third-hello
      taskRef:
        name: echo-hello-world
      params:
        - name: greeting
          value: "$(params.third-hello)"
      runAfter:
        - first-hello
      conditions:
        - conditionRef: do-extra-hello
          params:
            - name: do-extra-hello
              value: "$(params.do-extra-hello)"

---
apiVersion: tekton.dev/v1alpha1
kind: Condition
metadata:
  name: do-extra-hello
spec:
  params:
    - name: do-extra-hello
  check:
    image: debian:buster-slim
    script: '[ "$(params.do-extra-hello)" = "true" ]'

---
apiVersion: tekton.dev/v1beta1
kind: PipelineRun
metadata:
  name: hello-hello
spec:
  pipelineRef:
    name: hello-hello
  params:
    - name: first-hello
      value: asdf
    - name: second-hello
      value: zxcv
    - name: do-extra-hello
      value: "true"

```

--- a/pkg/cmd/pipelinerun/describe_test.go
+++ b/pkg/cmd/pipelinerun/describe_test.go
@@ -1070,3 +1070,145 @@ func TestPipelineRunDescribeV1beta1(t *testing.T) {
 	}
 	golden.Assert(t, got, fmt.Sprintf("%s.golden", t.Name()))
 }
+
+func TestPipelineRunDescribeV1beta1_taskrun_with_no_status(t *testing.T) {
+	clock := clockwork.NewFakeClock()
+	pipelinerunname := "pipeline-run"
+	taskRuns := []*v1beta1.TaskRun{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "ns",
+				Name:      "tr-1",
+				Labels:    map[string]string{"tekton.dev/task": "task-1"},
+			},
+			Spec: v1beta1.TaskRunSpec{
+				TaskRef: &v1beta1.TaskRef{
+					Name: "task-1",
+				},
+			},
+			Status: v1beta1.TaskRunStatus{
+				Status: duckv1beta1.Status{
+					Conditions: duckv1beta1.Conditions{
+						{
+							Status: corev1.ConditionFalse,
+							Reason: resources.ReasonFailed,
+						},
+					},
+				},
+				TaskRunStatusFields: v1beta1.TaskRunStatusFields{
+					StartTime:      &metav1.Time{Time: clock.Now()},
+					CompletionTime: &metav1.Time{Time: clock.Now().Add(5 * time.Minute)},
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "ns",
+				Name:      "tr-2",
+				Labels:    map[string]string{"tekton.dev/task": "task-1"},
+			},
+			Spec: v1beta1.TaskRunSpec{
+				TaskRef: &v1beta1.TaskRef{
+					Name: "task-1",
+				},
+			},
+		},
+	}
+
+	prun := []*v1beta1.PipelineRun{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pipelinerunname,
+				Namespace: "ns",
+			},
+			Spec: v1beta1.PipelineRunSpec{
+				PipelineRef: &v1beta1.PipelineRef{
+					Name: "pipeline",
+				},
+				Resources: []v1beta1.PipelineResourceBinding{
+					{
+						Name: "res-1",
+						ResourceRef: &v1beta1.PipelineResourceRef{
+							Name: "test-res",
+						},
+					},
+					{
+						Name: "res-2",
+						ResourceRef: &v1beta1.PipelineResourceRef{
+							Name: "test-res2",
+						},
+					},
+				},
+				Params: []v1beta1.Param{
+					{
+						Name: "p-1",
+						Value: v1beta1.ArrayOrString{
+							Type:      v1beta1.ParamTypeString,
+							StringVal: "somethingdifferent",
+						},
+					},
+					{
+						Name: "p-2",
+						Value: v1beta1.ArrayOrString{
+							Type:     v1beta1.ParamTypeArray,
+							ArrayVal: []string{"booms", "booms", "booms"},
+						},
+					},
+				},
+			},
+			Status: v1beta1.PipelineRunStatus{
+				Status: duckv1beta1.Status{
+					Conditions: duckv1beta1.Conditions{
+						{
+							Status:  corev1.ConditionTrue,
+							Reason:  resources.ReasonSucceeded,
+							Message: "Completed",
+						},
+					},
+				},
+				PipelineRunStatusFields: v1beta1.PipelineRunStatusFields{
+					StartTime:      &metav1.Time{Time: clock.Now()},
+					CompletionTime: &metav1.Time{Time: clock.Now().Add(20 * time.Minute)},
+					TaskRuns: map[string]*v1beta1.PipelineRunTaskRunStatus{
+						"tr-1": {
+							PipelineTaskName: "t-1",
+							Status:           &taskRuns[0].Status,
+						},
+						"tr-2": {
+							PipelineTaskName: "t-2",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	namespaces := []*corev1.Namespace{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "ns",
+			},
+		},
+	}
+
+	version := "v1beta1"
+	tdc := testDynamic.Options{}
+	dynamic, err := tdc.Client(
+		cb.UnstructuredV1beta1PR(prun[0], version),
+		cb.UnstructuredV1beta1TR(taskRuns[0], version),
+		cb.UnstructuredV1beta1TR(taskRuns[1], version),
+	)
+	if err != nil {
+		t.Errorf("unable to create dynamic client: %v", err)
+	}
+	cs, _ := test.SeedV1beta1TestData(t, pipelinev1beta1test.Data{Namespaces: namespaces, PipelineRuns: prun, TaskRuns: taskRuns})
+	cs.Pipeline.Resources = cb.APIResourceList(version, []string{"pipelinerun", "taskrun"})
+	p := &test.Params{Tekton: cs.Pipeline, Kube: cs.Kube, Dynamic: dynamic}
+
+	pipelinerun := Command(p)
+	got, err := test.ExecuteCommand(pipelinerun, "desc", "-n", "ns", pipelinerunname)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	golden.Assert(t, got, fmt.Sprintf("%s.golden", t.Name()))
+}

--- a/pkg/cmd/pipelinerun/testdata/TestPipelineRunDescribeV1beta1_taskrun_with_no_status.golden
+++ b/pkg/cmd/pipelinerun/testdata/TestPipelineRunDescribeV1beta1_taskrun_with_no_status.golden
@@ -1,0 +1,25 @@
+Name:           pipeline-run
+Namespace:      ns
+Pipeline Ref:   pipeline
+
+Status
+
+STARTED         DURATION     STATUS
+0 seconds ago   20 minutes   Succeeded
+
+Resources
+
+ NAME    RESOURCE REF
+ res-1   test-res
+ res-2   test-res2
+
+Params
+
+ NAME   VALUE
+ p-1    somethingdifferent
+ p-2    [booms booms booms]
+
+Taskruns
+
+ NAME   TASK NAME   STARTED         DURATION    STATUS
+ tr-1   t-1         0 seconds ago   5 minutes   Failed

--- a/pkg/pipelinerun/description/description.go
+++ b/pkg/pipelinerun/description/description.go
@@ -93,8 +93,9 @@ STARTED	DURATION	STATUS
  No taskruns
 {{- else }}
  NAME	TASK NAME	STARTED	DURATION	STATUS
-{{- range $taskrun := .TaskrunList }}
+{{- range $taskrun := .TaskrunList }}{{ if checkTRStatus $taskrun }}
  {{decorate "bullet" $taskrun.TaskrunName }}	{{ $taskrun.PipelineTaskName }}	{{ formatAge $taskrun.Status.StartTime $.Params.Time }}	{{ formatDuration $taskrun.Status.StartTime $taskrun.Status.CompletionTime }}	{{ formatCondition $taskrun.Status.Conditions }}
+{{- end }}
 {{- end }}
 {{- end }}
 `
@@ -109,11 +110,11 @@ type taskrunList []tkr
 func (trs taskrunList) Len() int      { return len(trs) }
 func (trs taskrunList) Swap(i, j int) { trs[i], trs[j] = trs[j], trs[i] }
 func (trs taskrunList) Less(i, j int) bool {
-	if trs[j].Status.StartTime == nil {
+	if trs[j].Status == nil || trs[j].Status.StartTime == nil {
 		return false
 	}
 
-	if trs[i].Status.StartTime == nil {
+	if trs[i].Status == nil || trs[i].Status.StartTime == nil {
 		return true
 	}
 
@@ -167,6 +168,7 @@ func PrintPipelineRunDescription(s *cli.Stream, prName string, p cli.Params) err
 		"pipelineResourceRefExists": validate.PipelineResourceRefExists,
 		"decorate":                  formatted.DecorateAttr,
 		"getTimeout":                getTimeoutValue,
+		"checkTRStatus":             checkTaskRunStatus,
 	}
 
 	w := tabwriter.NewWriter(s.Out, 0, 5, 3, ' ', tabwriter.TabIndent)
@@ -204,4 +206,8 @@ func getTimeoutValue(pr *v1beta1.PipelineRun) string {
 		return pr.Spec.Timeout.Duration.String()
 	}
 	return ""
+}
+
+func checkTaskRunStatus(taskRun tkr) bool {
+	return taskRun.PipelineRunTaskRunStatus.Status != nil
 }

